### ISSUE TITLE
A way to REPLY with Error when JSON.unmarshal failed in encoded connection

### ISSUE THREAD

**Issue by khanakia** [nats-io/nats.go#985]
A way to let the client know he sent the bad JSON instead of sending just a timeout in case of REQUEST-REPLY pattern
```
nc, err := nats.Connect(nats.DefaultURL()
c, err := nats.NewEncodedConn(nc, nats.JSON_ENCODER)
ec.Subscribe("auth.register", func(subj, reply string, msg *UserRequest) {
	if msg == nil {
		ec.Publish(reply, &ErrorResp{
	  	Code: "400",
	  	Msg:  "Empty user",
	  })	
	}
  
})
```

**SEND BAD JSON** using nat-cli
Run the command in terminal
```
nats request auth.register 'bad'
// Output: .
// Received empty response just a timeout
```

**Nats-Server Reported the error**
Got an error `nats: Got an error trying to unmarshal: invalid character 'b' looking for beginning of value auth.register`

Currently, i get a `request timeout error` on the client-side. How do reply with this error instead of timeout?

**Comment by khanakia** [comment]
Any update ?

**Comment by derekcollison** [comment]
@khanakia Thanks for the patience. The encoded connection layer was an early experiment and has not gotten much time or attention lately as we have been looking across all clients for a client v2 API that would probably include "middleware" layers similar to http stacks.

Have you looked here at all?

https://github.com/nats-io/nats.go/blob/main/enc_test.go#L92

**Comment by khanakia** [comment]
@derekcollison 
1. I checked the example but how do I send REPLY back to the client to let the client know there is an error instead of timeout. The middleware does not have the option to get the SUBJECT_ID for the client who sent the request so I can send the reply to the same ID.

2. As you said it was an early experiment. So will encoded connection still be supported in the future or should I change back to raw connection as I started to use NATS extensively in all of my applications?
